test/minikube/minikube_insights_test.go

```diff
@@ -28,13 +28,28 @@ func runYcsbForDurationAndScaleDown(t *testing.T, namespaceName string, options
 			ycsbNrReplicas = replicas
 		}
 	}
-	testlib.AwaitNrReplicasScheduled(t, namespaceName, YCSB_CONTROLLER_NAME, ycsbNrReplicas)
-	testlib.AwaitNrReplicasReady(t, namespaceName, YCSB_CONTROLLER_NAME, ycsbNrReplicas)
-	time.Sleep(delay)
-	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
-	// Ignore the error as at the time of execution, the controller may be removed by the test tear down
-	k8s.RunKubectlE(t, kubectlOptions, "scale", "replicationcontroller", YCSB_CONTROLLER_NAME, "--replicas=0")
-	t.Log("YCSB workload stopped")
+
+	if ycsbNrReplicas > 0 {
+		kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+
+		testlib.AddDiagnosticTeardown(testlib.TEARDOWN_YCSB, t, func() {
+			_ = k8s.RunKubectlE(t, kubectlOptions, "describe", "replicationcontroller", YCSB_CONTROLLER_NAME)
```

Comment:
Do you need this assignment statement and the one below to make the compiler happy?